enhancements/installer/assisted-installer-metrics.md

```diff
@@ -0,0 +1,398 @@
+---
+title: assisted-installer-telemetry-metrics
+authors:
+  - "@atraeger"
+reviewers:
+  - TBD
+  - "@alicedoe"
+approvers:
+  - TBD
+  - "@oscardoe"
+creation-date: 2020-07-22
+last-updated: 2020-07-22
+status: implementable
+see-also:
+  - "/enhancements/installer/connected-assisted-installer.md"
+---
+
+# Assisted Installer Telemetry Metrics
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The assisted installer, [previously proposed and accepted](connected-assisted-installer.md), collects informative installation metrics. This enhancement proposes that the assisted installer would push metrics related to service usage, user workflows, and installation statistics. This enhancement focuses on the design of the metrics to be pushed to Prometheus.
+
+## Motivation
+
+Metrics from the assisted installer service would help engineers and product managers to understand different user workflows, environments, experiences, difficulties, and feature usage. This would benefit not only future versions of the assisted installer but also the OpenShift installer itself.
+
+### Goals
+
+1. Describe a set of metrics to fulfill the user stories described below, while keeping cardinality reasonable.
+2. Describe how the metrics will be pushed to Prometheus.
+
+### Non-Goals
+
+1. Collected metrics will not allow analysis of individual installations.
+
+## Proposal
+
+### User Stories
+
+#### Telemetry Analyst
```

Comment:
I'm missing the following:
- private dns / hosted dns
- user defined vips / autoallocated (not existing yet, I know)
- I really want to see breakage of platform time - vms vs real bare metal since it will affect the timings of course
- no metrics about iso - new /reused, time to generate

Also:
As a telemetry analyst, I would like to understand what are the phases that causes most of the installation failures (for example, hosts does not recover from rebooting)

Reply:
Most of the LE start with irrelevant numbers  - 0.005 - not sure that any step can take this time. Lets do some cleanup

Reply:
> I'm missing the following:
> 
>     * private dns / hosted dns

Added counter for hosted vs. private DNS

>     * user defined vips / autoallocated (not existing yet, I know)

Will add when we will support

>     * I really want to see breakage of platform time - vms vs real bare metal since it will affect the timings of course

Do we want to add a platform label to every metric?  Probably...

>     * no metrics about iso - new /reused, time to generate

ISO creation is synchronous, we have API latencies

> Also:
> As a telemetry analyst, I would like to understand what are the phases that causes most of the installation failures (for example, hosts does not recover from rebooting)

Added

Reply:
> Most of the LE start with irrelevant numbers - 0.005 - not sure that any step can take this time. Lets do some cleanup

I already changed the histogram values in this document.